### PR TITLE
Decreasing SPIFFS partition size by 4 bytes to workaround issue #7467

### PR DESCRIPTION
**Pull request by CoryCharlton**
## Description of Change
Decreasing SPIFFS partition size by 1KB to workaround issue #7467

## Tests scenarios
Tested with PIO using Arduino Core v2.0.5 using an Unexpected Maker FeatherS3

## Related links
Works around #7467

### CODE REVIEW

File: tools/partitions/default_16MB.csv [hunk 1]
```diff
@@ -3,4 +3,4 @@ nvs,      data, nvs,     0x9000,  0x5000,
 otadata,  data, ota,     0xe000,  0x2000,
 app0,     app,  ota_0,   0x10000, 0x640000,
 app1,     app,  ota_1,   0x650000,0x640000,
-spiffs,   data, spiffs,  0xc90000,0x370000,
+spiffs,   data, spiffs,  0xc90000,0x36FC00,
```
Comment (SuGlider) [inline tools/partitions/default_16MB.csv:6]: @CoryCharlton - please test it with:

spiffs, data, spiffs, 0xc90000,0x36FFFC,

File: tools/partitions/large_spiffs_16MB.csv [hunk 2]
```diff
@@ -3,4 +3,4 @@ nvs,      data, nvs,     0x9000,  0x5000,
 otadata,  data, ota,     0xe000,  0x2000,
 app0,     app,  ota_0,   0x10000, 0x480000,
 app1,     app,  ota_1,   0x490000,0x480000,
-spiffs,   data, spiffs,  0x910000,0x6F0000,
+spiffs,   data, spiffs,  0x910000,0x6EFC00,
```
Comment (SuGlider) [inline tools/partitions/large_spiffs_16MB.csv:6]: @CoryCharlton - please test it with:

spiffs, data, spiffs, 0x910000,0x6EFFFC,

**Review by SuGlider — CHANGES_REQUESTED** [review]

File: tools/partitions/large_spiffs_16MB.csv (hunk 2 above)
Comment (CoryCharlton) [reply tools/partitions/large_spiffs_16MB.csv]: Confirm they work and committed the updated files.

**Comment by BlueAndi** [comment]
@SuGlider, @CoryCharlton The 4 byte less corresponds in a filesystem which is not 4 k aligned. Did you try an OTA update of the filesystem? 

https://github.com/espressif/arduino-esp32/blob/972c3bbd00e7294739ba9b5eea0db3a0298d0b53/libraries/Update/src/Updater.cpp#L153-L158
https://github.com/espressif/arduino-esp32/blob/972c3bbd00e7294739ba9b5eea0db3a0298d0b53/libraries/Update/src/Updater.cpp#L207-L215
https://github.com/espressif/arduino-esp32/blob/972c3bbd00e7294739ba9b5eea0db3a0298d0b53/libraries/Update/src/Update.h#L32

Means this must be considered:
```c
#define SPI_FLASH_SEC_SIZE  4096    /**< SPI Flash sector size */
```

**Comment by Jason2866** [comment]
Is this still needed with the change/fix of esptool commit? https://github.com/espressif/esptool/tree/21e59148e73e384947327a0883a753685e03d143

**Comment by SuGlider** [comment]
> Is this still needed with the change/fix of esptool commit? https://github.com/espressif/esptool/tree/21e59148e73e384947327a0883a753685e03d143

I think that it may have fixed the issue reported by @CoryCharlton 
@CoryCharlton - could you please test the latest version of esptool to check if the issue you reported is solved?
Thanks.

**Comment by SuGlider** [comment]
@me-no-dev - not sure how this fix to esptool may be reflected into the Arduino Core tools. 
Would it be necessary to update some files?

**Comment by me-no-dev** [comment]
@SuGlider this fix needs to be released from esptool team and then we can try to update it for Arduino

**Comment by Jason2866** [comment]
@me-no-dev  Not using Arduino IDE so maybe a total dumb question. Why not using esptool.py directly? You are using in Arduino Lib Builder components from latest master branch to build the libs. Why not doing for esptool.py too?

**Comment by CoryCharlton** [comment]
> I think that it may have fixed the issue reported by @CoryCharlton
@CoryCharlton - could you please test the latest version of esptool to check if the issue you reported is solved?
Thanks.

Confirmed that this change resolves the issue. Closing this PR and will close the issue.

**Comment by me-no-dev** [comment]
> @me-no-dev Not using Arduino IDE so maybe a total dumb question. Why not using esptool.py directly? You are using in Arduino Lib Builder components from latest master branch to build the libs. Why not doing for esptool.py too?

we pull esptool when released (or when it's necessary), but in this case, the change that fixes this issue is not part of any released esptool yet. We can not use the one in ESP-IDF, because for Windows and Mac we need the compiled versions (which are part of the releases of esptool)

**Pull request closed without merging**